Add zero count outcomes (#2889)
PennyLaneAI/pennylane: doc/introduction/measurements.rst

    def circuit():
        qml.Hadamard(wires=0)
        qml.CNOT(wires=[0, 1])
        return qml.counts()

And the result is:
           
>>> result = circuit()
>>> print(result)
{'00': 495, '11': 505}

> Per default, only observed outcomes are included in the dictionary. The kwarg all_outcomes=True can 

Reviewer: ```suggestion
Per default, only observed outcomes are included in the dictionary. The kwarg ``all_outcomes=True`` can 
```

> be used to display all possible outcomes, including those that were observed 0 times in sampling.

Reviewer: ```suggestion
be used to display all possible outcomes, including those that were observed ``0`` times in sampling.
```

> For example, we could run the previous circuit with all_outcomes=True:

Reviewer: ```suggestion
For example, we could run the previous circuit with ``all_outcomes=True``:
```

.. code-block:: python

dev = qml.device("default.qubit", wires=2, shots=1000)

@qml.qnode(dev)
def circuit():
    qml.Hadamard(wires=0)
    qml.CNOT(wires=[0, 1])
    return qml.counts(all_outcomes=True)

>>> result = circuit()
>>> print(result)
{'00': 518, '01': 0, '10': 0, '11': 482}

> Note: For complicated Hamiltonians, this can add considerable overhead time (due to cost of calcuating 

Reviewer: ```suggestion
Note: For complicated Hamiltonians, this can add considerable overhead time (due to the cost of calculating 
```

> eigenvalues to determine possible outcomes), and as number of qubits increases, the length of the output 

Reviewer: ```suggestion
eigenvalues to determine possible outcomes), and as the number of qubits increases, the length of the output 
```

dictionary showing possible computational basis states grows rapidly. 

> 
> 

Reviewer: ```suggestion
```

>>> circuit()
{'00': 495, '11': 505}

Per default, only observed outcomes are included in the dictionary. The kwarg ``all_outcomes=True`` can 
be used to display all possible outcomes, including those that were observed ``0`` times in sampling.

For example, we could run the previous circuit with ``all_outcomes=True``:

.. code-block:: python

dev = qml.device("default.qubit", wires=2, shots=1000)

@qml.qnode(dev)
def circuit():
    qml.Hadamard(wires=0)
    qml.CNOT(wires=[0, 1])

>     return qml.counts(all_outcomes=True)

Reviewer 2: You need to indent the code block.

    def circuit():
        qml.Hadamard(wires=0)
        qml.CNOT(wires=[0,1])

>         qml.PauliX(wires=1)

Reviewer 2: Remove third qubit of the output of `print(result)`:

```pycon
[tensor(0.026, requires_grad=True) tensor(0.026, requires_grad=True)
 tensor({'01': 513, '11': 487}, dtype=object, requires_grad=True)]
```

Moreover, I get a tuple instead of a tensor as output. @antalszava was this changed in the return types PR?

>         qml.PauliX(wires=1)

Reviewer: No, it was reworked in https://github.com/PennyLaneAI/pennylane/pull/2839, but this output was missed there. Thanks for catching it!

>         qml.PauliX(wires=1)

Reviewer 2: @lillian542 could you change the output to a tuple too?

```pycon
(0.026, 0.026, {'01': 513, '11': 487})
```
